cost_index/utils.py

```diff
@@ -29,3 +32,74 @@ def adjust_value(value: Decimal, start_index: Decimal, end_index: Decimal):
 
     # Round to floor 2 decimals
     return Decimal(math.floor(adjusted_value * 100)) / 100
+
+
+def current_right_of_occupancy_payment(
+    apartment_uuid, original_right_of_occupancy_payment, before: date = None
+):
+    """
+    Return current right of occupancy payment in cents
+    """
+    revaluation_qs = ApartmentRevaluation.objects.filter(
+        apartment_reservation__apartment_uuid=apartment_uuid
+    ).order_by("-end_date")
+
+    if before:
+        revaluation_qs = revaluation_qs.filter(end_date__lte=before)
```

Comment:
"before" seems a tiny bit misleading term if also the same day is included?

Reply:
`on_or_before` vs `not_after`, I guess I'll go with the latter